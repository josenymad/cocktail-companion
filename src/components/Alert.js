import React from "react";
import "../styles/alert.css";

const Alert = ({ alert }) => {
  if (!alert) return null;
  return <div className="alert">{alert}</div>;
};

export default Alert;
