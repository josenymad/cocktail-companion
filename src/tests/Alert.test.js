import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert", () => {
  it("displays an error message", () => {
    const { getByText, asFragment } = render(
      <Alert alert="Something went wrong" />,
    );

    expect(getByText(/Something went wrong/).textContent).toBe(
      "Something went wrong",
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("does not display a message if the alert prop is an empty string", () => {
    const { asFragment } = render(<Alert alert="" />);
    const alert = asFragment();

    expect(alert).toMatchSnapshot();
  });
});
