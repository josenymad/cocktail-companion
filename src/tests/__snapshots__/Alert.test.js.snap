// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert displays an error message 1`] = `
<DocumentFragment>
  <div
    class="alert"
  >
    Something went wrong
  </div>
</DocumentFragment>
`;

exports[`Alert does not display a message if the alert prop is an empty string 1`] = `<DocumentFragment />`;
